CMakeLists.txt

```diff
@@ -1438,6 +1438,71 @@ if(WIN32)
+if(DJINTEROP)
+  message(STATUS "Enabling djinterop support...")
+  target_compile_definitions(mixxx-lib PUBLIC __DJINTEROP__)
+
+  # Look for an existing installation of libdjinterop and use that if available.
+  # Otherwise, download and build from GitHub.
+  find_package(DjInterop)
+  if(DjInterop_FOUND)
+    # An existing installation of djinterop is available.
+    message(STATUS "Linking dynamically to existing installation of libdjinterop")
+    target_include_directories(mixxx-lib PUBLIC ${DjInterop_INCLUDE_DIRS})
+    target_link_libraries(mixxx-lib PUBLIC DjInterop::DjInterop)
+  else()
+    # Fetch djinterop sources from GitHub and build them statically.
+    message(STATUS "Linking statically to libdjinterop fetched from GitHub")
+
+    # Meson is the preferred build system for libdjinterop, but may not be
+    # available to all developers wishing to merely compile Mixxx.  However,
+    # libdjinterop provides a minimal CMake build too for use with in-tree
+    # builds like this one, which we will default to.
+    option(DJINTEROP_MESON_BUILD "Build djinterop library with meson" OFF)
+
+    set(DJINTEROP_GIT_REPO_URL https://github.com/xsco/libdjinterop.git)
+    set(DJINTEROP_GIT_TAG tags/0.11.0)
+    set(DJINTEROP_INSTALL_DIR "${CMAKE_CURRENT_BINARY_DIR}/lib/libdjinterop-install")
+    if(DJINTEROP_MESON_BUILD)
+      # Note this requires meson 0.54 or later.
+      ExternalProject_Add(libdjinterop
+        GIT_REPOSITORY ${DJINTEROP_GIT_REPO_URL}
+        GIT_TAG ${DJINTEROP_GIT_TAG}
+        CONFIGURE_COMMAND meson setup --prefix ${DJINTEROP_INSTALL_DIR} --libdir lib --includedir include build <SOURCE_DIR>
+        BUILD_COMMAND meson compile -C build
+        INSTALL_COMMAND meson install -C build
+        EXCLUDE_FROM_ALL TRUE
+      )
+    else()
+      ExternalProject_Add(libdjinterop
+        GIT_REPOSITORY ${DJINTEROP_GIT_REPO_URL}
+        GIT_TAG ${DJINTEROP_GIT_TAG}
+        INSTALL_DIR ${DJINTEROP_INSTALL_DIR}
+        CMAKE_ARGS "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}" -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
+        EXCLUDE_FROM_ALL TRUE
+      )
+    endif()
+
+    # Assemble a library based on the external project.
+    add_library(mixxx-libdjinterop STATIC IMPORTED)
+    add_dependencies(mixxx-libdjinterop libdjinterop)
+    set(DJINTEROP_INCLUDE_DIR "${DJINTEROP_INSTALL_DIR}/include")
+    set(DJINTEROP_LIBRARY "lib/${CMAKE_STATIC_LIBRARY_PREFIX}djinterop${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    set(DJINTEROP_LIBRARY_DIR "${DJINTEROP_INSTALL_DIR}/${DJINTEROP_LIBRARY}")
+    set_target_properties(mixxx-libdjinterop PROPERTIES IMPORTED_LOCATION "${DJINTEROP_LIBRARY_DIR}")
+    target_include_directories(mixxx-lib PUBLIC ${DJINTEROP_INCLUDE_DIR})
+    target_link_libraries(mixxx-lib PUBLIC mixxx-libdjinterop)
+
+    # libdjinterop depends on zlib and sqlite3.
+    # Mixxx already has a dependency on sqlite3, but not zlib, which we add now.
+    find_package(ZLIB 1.2.8 REQUIRED)
```

Comment:
This might be considered slightly fishy: we're linking statically to libdjinterop, but libdjinterop depends on SQLite3 and ZLib.  It just so happens that Mixxx already has the SQLite3 dependency, but not ZLib.

Are there any concerns with encoding implicit knowledge of libdjinterop's dependencies here?  I'm not aware of a better way with  static build.

Reply:
@Holzhaus Opinions?

Reply:
I think this should be move to the FindDJInterop module. For the external project build, this can be taken care of in libdjinterop's own CMakeLists.txt.

Reply:
> For the external project build, this can be taken care of in libdjinterop's own CMakeLists.txt.

The [CMakeLists.txt file from libdjinterop](https://github.com/xsco/libdjinterop/blob/6b73cd37bf63f0a1fdbba1652cf8b696f4e38572/CMakeLists.txt#L19) already has a `find_package(ZLIB 1.2.8 REQUIRED)` line.  However, I was not able to get the dependencies of the external project to transfer through to `mixxx-lib` (and search engines were not fruitful for me).  Do you know of a way in CMake to do this?

Technically, the line isn't needed in the `FindDjInterop` because it would link against the shared library, which already has the ZLib dependency.  At the very least (if deps can't be inherited), this is only needed for the static build.

Reply:
I just noticed that the ChromaPrint dependency already does this (with FFTW), so at least the situation here is in line with the status quo.